### PR TITLE
chore: fix the comment describing the magic of updateOpaqueInstance

### PR DESCRIPTION
**Pull request by glennsl**
I'm not entirely confident that this is correct, so consider it more of a suggestion, but it should at least be more accurate than before!

### CODE REVIEW

**Comment by glennsl** [comment]
> I have updated your lock dirs and formatted the code. Please @glennsl pull the last commit before pushing any more changes.

Very good, sir. Thank you for your assistance.

File: lib/Brisk_reconciler_internal.re [hunk 1]
```diff
@@ -661,20 +661,16 @@ module Render = {
...
-      * This approach is as sound as our confidence in our ability to repair the
-      * mutations accurately.
-      *
-      * There are more elegant ways to do this using first class modules, combined
-      * with extensible variants. That is how we should do this if we want to turn
-      * this implementation into something serious - it would avoid hitting the
-      * write barrier.
+      * For details, see:
+      *   https://discuss.ocaml.org/t/types-as-first-class-citizens-in-ocaml/2030/3
+      *   https://alan.petitepomme.net/cwn/2015.03.24.html#1
```
Comment (wokalski) [inline lib/Brisk_reconciler_internal.re:673]: Very nice. Thank you @glennsl. Excellent.

File: lib/Brisk_reconciler_internal.re [hunk 2]
```diff
@@ -661,20 +661,16 @@ module Render = {
...
-      * different 'x. We need to see if the 'x are the same and if so safely "cast"
-      * one's `component` to the others. We do this by handing off state safely into
-      * one of their `component`s, and then seeing if we can observe it in the
-      * other, and if so, we simply treat the old component as the new one's.
+      * a type witness! We have a value of type `instance` and another of type
+      * `element`, where each has their own `component 'x` for potentially
+      * different 'x. We need to see if the they are the same and if so safely "cast"
+      * one's `component` to the others. We do this by using a type witness that is
+      * able to prove type equality to the compiler through the use of an extensible
+      * GADT as a kind of dynamic type.
```
Comment (wokalski) [inline lib/Brisk_reconciler_internal.re:669]:
```suggestion
      * a type witness! We have a value of type `instance` and another of type
      * `element`, where each has their own `component 'x` with potentially
      * different type variable 'x. We need to see if they are the same and if so
      * safely "cast" one of the components. We do this by using a type witness that is
      * able to prove type equality to the compiler through the use of an extensible
      * GADT as a kind of dynamic type.
```

**Review by wokalski — APPROVED** [review]
This has been much needed. I also added a small optional tweak to the explanation. Feel free to discard them.

File: lib/Brisk_reconciler_internal.re (hunk 2 above)
Comment (glennsl) [reply lib/Brisk_reconciler_internal.re:669]: LGTM. Thanks!

**Pull request merged**